src/Equinor.ProCoSys.IPO.Command/InvitationCommands/FillRfocStatuses/FillRfocStatusesCommandHandler.cs

```diff
@@ -0,0 +1,152 @@
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.ProjectAggregate;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.CommPkg;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.McPkg;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Command.InvitationCommands.FillRfocStatuses
+{
+    public class FillRfocStatusesCommandHandler : IRequestHandler<FillRfocStatusesCommand, Result<Unit>>
+    {
+        private readonly IInvitationRepository _invitationRepository;
+        private readonly IProjectRepository _projectRepository;
+        private readonly IMcPkgApiService _mcPkgApiService;
+        private readonly ICommPkgApiService _commPkgApiService;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly ILogger<FillRfocStatusesCommandHandler> _logger;
+
+        public FillRfocStatusesCommandHandler(
+            IInvitationRepository invitationRepository,
+            IUnitOfWork unitOfWork,
+            ILogger<FillRfocStatusesCommandHandler> logger,
+            IProjectRepository projectRepository,
+            IMcPkgApiService mcPkgApiService,
+            ICommPkgApiService commPkgApiService)
+        {
+            _invitationRepository = invitationRepository;
+            _logger = logger;
+            _unitOfWork = unitOfWork;
+            _projectRepository = projectRepository;
+            _mcPkgApiService = mcPkgApiService;
+            _commPkgApiService = commPkgApiService;
+        }
+
+        public async Task<Result<Unit>> Handle(FillRfocStatusesCommand request, CancellationToken cancellationToken)
+        {
+            // THIS CODE WAS WRITTEN TO RUN A ONETIME TRANSFORMATION WHEN WE INTRODUCED RfocAccepted on McPkgs and CommPkgs,
+            // and new status (ScopeHandedOver) on invitation
+            // WE KEEP THE CODE ... MAYBE WE WANT TO DO SIMILAR STUFF LATER
+
+            var allProjects = await _projectRepository.GetAllAsync();
+            var invitations = await _invitationRepository.GetAllAsync();
+
+            var mcPkgsUpdatedCount = 0;
+            var commPkgsUpdatedCount = 0;
+            var invitationsSetToScopeHandedOverCount = 0;
+
+            foreach (var project in allProjects)
+            {
+                var invitationsInProject = invitations.Where(i => i.ProjectId == project.Id).ToList();
+
+                mcPkgsUpdatedCount += await HandleMcPkgsAsync(invitationsInProject, project, mcPkgsUpdatedCount);
+                commPkgsUpdatedCount += await HandleCommPkgsAsync(invitationsInProject, project, commPkgsUpdatedCount);
+                invitationsSetToScopeHandedOverCount += HandleInvitationStatus(invitationsInProject, invitationsSetToScopeHandedOverCount);
+                _logger.LogInformation($"FillRfocStatuses: Project updated: {project.Name}");
+            }
+
+            if (request.SaveChanges && (mcPkgsUpdatedCount > 0 || commPkgsUpdatedCount > 0))
```

Comment:
You should splitt the if:
![image](https://github.com/equinor/procosys-invitation-for-punch-out-api/assets/19545528/128994b3-3ac5-4db4-aca9-46c1a78c439d)